### PR TITLE
Build snippets during Analyze job by default

### LINKED ISSUE #33877: Change default BuildSnippets to check all snippets are used
The CI pipeline can check that all snippets are used to make sure people don't accidentally forget to use any they defined (and also makes sure any referenced are defined), but the default is turned off currently. Many SDKs have already opted in to do this check during the Analyze job, but I think we should enable that by default.

Running `eng/scripts/Update-Snippets.ps1 * -StrictMode`, I find the following SDKs still have some unreferenced snippets:

- [ ] communication
- [ ] dns
- [ ] formrecognizer
- [ ] iot
- [ ] servicebus

We should make sure that a new instance of a DPG template does not run afoul of this error as well, but given the command above didn't find any issues it should be fine. We should create an instance though and double check.

Client library authors should run `Update-Snippets.ps1 {ServiceDirectory}` to check compliance, and to make sure any README or sample markdowns are properly updated (which the Analyze step will also do or fail).

Also see related snippet clean-up issue #20431

### PR DESCRIPTION
**Pull request by heaths**
Resolves #33877

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/iot/ci.yml [hunk 1]
```diff
@@ -20,8 +20,9 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
   parameters:
     ServiceDirectory: iot
+    BuildSnippets: true
```
Comment (jsquire) [inline sdk/iot/ci.yml:23]: If this is being done by default, do we need this here?   Maybe it was intended to be `false`?

File: sdk/dns/ci.mgmt.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
   parameters:
     ServiceDirectory: dns
+    BuildSnippets: true
```
Comment (jsquire) [inline sdk/dns/ci.mgmt.yml:21]: Since this is done by default, do we need this here?

**Review by jsquire — APPROVED** [review]

**Review by annelo-msft — APPROVED** [review]
ACR LGTM, Thanks @heaths!

**Review by weshaggard — APPROVED** [review]
The change looks reasonable to me.

File: sdk/iot/ci.yml (hunk 1 above)
Comment (heaths) [reply sdk/iot/ci.yml:23]: `true`, `false` - you say "toe-ma-toe" I say "toe-may-toe".

But, yes. :)

**Comment by pallavit** [comment]
Do we want to create follow-up issues for the SDKs where we had to manually set this to false? This looks great, thank you for doing it.

**Review by pallavit — APPROVED** [review]

**Comment by heaths** [comment]
> Do we want to create follow-up issues for the SDKs where we had to manually set this to false? This looks great, thank you for doing it.

@pallavit already did. :) See https://github.com/Azure/azure-sdk-for-net/issues/20431. I updated the OP with individual tracking issues as tasks.

**Pull request merged**